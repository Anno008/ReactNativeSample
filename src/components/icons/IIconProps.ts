export interface IIconProps {
    tintColor: string;
    size: number;
}
