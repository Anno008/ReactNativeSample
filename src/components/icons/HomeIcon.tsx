import React from "react";
import { View } from "react-native";
import { Path, Svg } from "react-native-svg";
import { IIconProps } from "./IIconProps";

export const HomeIcon: React.SFC<IIconProps> = (props: IIconProps): JSX.Element => (
    <View>
        <Svg width={props.size} height={props.size} viewBox="0 0 24 24">
            <Path fill="none" d="M0 0h24v24H0V0z" />
            <Path fill={props.tintColor} d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8h5z" />
        </Svg>
    </View>
);
