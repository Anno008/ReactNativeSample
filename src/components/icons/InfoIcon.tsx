import React from "react";
import { View } from "react-native";
import { Path, Svg } from "react-native-svg";
import { IIconProps } from "./IIconProps";

export const InfoIcon: React.SFC<IIconProps> = (props: IIconProps) => (
  <View>
    <Svg width={props.size} height={props.size} viewBox="0 0 24 24">
      <Path fill="none" d="M0 0h24v24H0V0z" />
      <Path
        fill={props.tintColor}
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </Svg>
  </View>
);
