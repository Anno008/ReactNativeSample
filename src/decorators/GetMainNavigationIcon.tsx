import { IIconProps } from "components/icons/IIconProps";
import React from "react";
import { TabScene } from "react-navigation";

export const getMainNavigationIcon = (
    WrappedComponent: React.SFC<IIconProps>,
    tabIconInfo: TabScene
): JSX.Element => (
    <WrappedComponent
        size={tabIconInfo.focused ? 30 : 24}
        tintColor={tabIconInfo.tintColor || "gray"}
    />
);
