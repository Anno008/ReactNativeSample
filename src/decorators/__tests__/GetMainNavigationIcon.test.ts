import { HomeIcon } from "components/icons/HomeIcon";
import { getMainNavigationIcon } from "decorators/GetMainNavigationIcon";
import { shallow } from "enzyme";
import { View } from "react-native";
import { Path } from "react-native-svg";
import { NavigationRoute, TabScene } from "react-navigation";

describe("GetMainNavigationIcon tests", () => {
  it("Should render the provided icon", () => {
    const mockProps: TabScene = {
      focused: false,
      index: 1,
      route: {} as NavigationRoute,
      tintColor: "blue"
    };
    const wrapper = shallow(getMainNavigationIcon(HomeIcon, mockProps));

    expect(wrapper).toMatchSnapshot();
  });

  it("Should render icon with width and height set to 30 if focused", () => {
    const mockProps: TabScene = {
      focused: true,
      index: 1,
      route: {} as NavigationRoute,
      tintColor: "blue"
    };

    const wrappedIcon = getMainNavigationIcon(HomeIcon, mockProps);
    const wrapper = shallow(wrappedIcon);

    const { height, width } = wrapper
      .find(View)
      .first()
      .getElement().props.children.props;

    expect(height).toEqual(30);
    expect(width).toEqual(30);
  });

  it("Should render icon with width and height set to 24 if not focused", () => {
    const mockProps: TabScene = {
      focused: false,
      index: 1,
      route: {} as NavigationRoute,
      tintColor: "blue"
    };

    const wrappedIcon = getMainNavigationIcon(HomeIcon, mockProps);
    const wrapper = shallow(wrappedIcon);

    const { height, width } = wrapper
      .find(View)
      .first()
      .getElement().props.children.props;

    expect(height).toEqual(24);
    expect(width).toEqual(24);
  });

  it("Should render icon with provided tint color if it exists in props", () => {
    const mockProps: TabScene = {
      focused: false,
      index: 1,
      route: {} as NavigationRoute,
      tintColor: "blue"
    };

    const wrappedIcon = getMainNavigationIcon(HomeIcon, mockProps);
    const wrapper = shallow(wrappedIcon);

    const { fill } = wrapper
      .find(Path)
      .get(1)
      .props;

    expect(fill).toEqual(mockProps.tintColor);
  });

  it("Should render icon with fallback tint color if it does not exists in props", () => {
    const mockProps: TabScene = {
      focused: false,
      index: 1,
      route: {} as NavigationRoute,
    };

    const wrappedIcon = getMainNavigationIcon(HomeIcon, mockProps);
    const wrapper = shallow(wrappedIcon);

    const { fill } = wrapper
      .find(Path)
      .get(1)
      .props;

    expect(fill).toEqual("gray");
  });
});
