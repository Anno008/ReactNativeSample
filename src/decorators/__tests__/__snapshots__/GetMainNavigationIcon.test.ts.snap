// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetMainNavigationIcon tests Should render the provided icon 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    size={24}
    tintColor="blue"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Svg
        height={24}
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
        width={24}
      >
        <Path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <Path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8h5z"
          fill="blue"
        />
      </Svg>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Path
            d="M0 0h24v24H0V0z"
            fill="none"
          />,
          <Path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8h5z"
            fill="blue"
          />,
        ],
        "height": 24,
        "preserveAspectRatio": "xMidYMid meet",
        "viewBox": "0 0 24 24",
        "width": 24,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "d": "M0 0h24v24H0V0z",
            "fill": "none",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "d": "M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8h5z",
            "fill": "blue",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Svg
          height={24}
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
          width={24}
        >
          <Path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <Path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8h5z"
            fill="blue"
          />
        </Svg>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Path
              d="M0 0h24v24H0V0z"
              fill="none"
            />,
            <Path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8h5z"
              fill="blue"
            />,
          ],
          "height": 24,
          "preserveAspectRatio": "xMidYMid meet",
          "viewBox": "0 0 24 24",
          "width": 24,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "d": "M0 0h24v24H0V0z",
              "fill": "none",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "d": "M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8h5z",
              "fill": "blue",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
